internal/impl/sftp/input.go

```diff
@@ -174,11 +174,62 @@ func newSFTPReaderFromParsed(conf *service.ParsedConfig, mgr *service.Resources)
 }
 
 func (s *sftpReader) Connect(ctx context.Context) (err error) {
+	file, nextPath, skip, err := s.seekNextPath(ctx)
+	if err != nil {
+		return err
+	}
+	if skip {
+		return nil
+	}
+
+	details := service.NewScannerSourceDetails()
+	details.SetName(nextPath)
+	if s.scanner, err = s.scannerCtor.Create(file, func(ctx context.Context, aErr error) (outErr error) {
```

Comment:
I think the assignment of `s.scanner` needs to be under the mutex as well based on the `ReadBatch` function right?